src/Libraries/RevitNodes/Elements/FamilyInstance.cs

```diff
@@ -330,28 +330,30 @@ public override string ToString()
         /// <returns>The result family instance.</returns>
         public FamilyInstance SetRotation(double degree)
         {
-           if (this == null)
-              throw new ArgumentNullException("familyInstance");
+            if (this == null)
+                throw new ArgumentNullException("familyInstance");
 
-           TransactionManager.Instance.EnsureInTransaction(Document);
+            TransactionManager.Instance.EnsureInTransaction(Document);
 
-           // Rotate the element.
-           var oldTransform = InternalFamilyInstance.GetTransform();
-           double[] oldRotationAngles;
-           TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
+            Document.Regenerate();
```

Comment:
I'll suggest, let's have an internal method to GetTransform() or GetRotation() and call Document.Regenerate() from there, so that we can reuse the same code later on.